lumicks/pylake/detail/image.py

```diff
@@ -129,7 +129,40 @@ def seek_timestamp_next_line(infowave):
     return pixel_start[idx + 1]
 
 
-def reconstruct_image_sum(data, infowave, pixels_per_line, lines_per_frame=None):
+def process_infowave(infowave):
+    # Example infowave:
+    #  1 0 0 1 0 1 2 0 1 0 0 1 0 1 0 1 0 0 1 0 2 0 1 0 1 0 0 1 0 1 0 1 2 1 0 0 1
+    #              ^ <-----------------------> ^                       ^
+    #                       one pixel
+    mask = infowave != InfowaveCode.discard
+    subset = infowave[mask]
+
+    # After discard:
+    #  1 1 1 2 1 1 1 1 1 2 1 1 1 1 1 2 1 1 1
+    #        ^ <-------> ^           ^
+    #         pixel_size (i.e. data samples per pixel)
+    # This should be:
+    #   pixel_sizes = np.diff(np.flatnonzero(infowave == InfowaveCode.pixel_boundary))
+    # But for now we assume that every pixel consists of the same number of samples
+    return subset, mask
+
+
+def round_up(size, n):
+        """Round up `size` to the nearest multiple of `n`"""
+        return int(math.ceil(size / n)) * n
+
+
+def reshape_reconstructed_image(pixels, shape):
+    # shape = (<<lines_per_frame>>, pixels_per_line)
+    resized_pixels = np.zeros(round_up(pixels.size, np.prod(shape)))
+    resized_pixels[:pixels.size] = pixels
+    if len(shape) > 1:
+        return resized_pixels.reshape(-1, *shape).squeeze()
+    else:
+        return resized_pixels.reshape(-1, *shape)
+
+
+def reconstruct_image_sum(data, infowave, shape):
     """Rapidly reconstruct a scan or kymograph image from raw data summing pixels where the infowave equates to 1.
     See reconstruct_image for more information.
```

Comment:
I should have added this at the time, but given that you're working on this code now, I think it wouldn't hurt to add a little warning to this function `reconstruct_image_sum` mentioning that this one should not be used for timestamps since they will overflow.